azurerm_postgresql_database.testdb: Code="InternalServerError" while destroying

### Terraform (and AzureRM Provider) Version
Terraform -v = 0.11.8
azurerm: version = "~> 1.30"

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_postgresql_database`

### Terraform Configuration Files

`{
  "resource" : {
    "azurerm_postgresql_server" : {
      "pgserver" : {
        "ssl_enforcement" : "Enabled",
        "administrator_login" : "${var.serveruser}",
        "administrator_login_password" : "${var.serverpassword}",
        "name" : "${var.pgservers[2]}",
        "storage_profile" : {
          "storage_mb" : "${var.storage_mb}",
          "backup_retention_days" : "${var.backup_retention_days}",
          "geo_redundant_backup" : "${var.geo_redundant_backup}"
        },
        "resource_group_name" : "${var.rgName}",
        "location" : "${var.location}",
        "sku" : {
          "family" : "${var.skufamily}",
          "name" : "${var.skuname}",
          "capacity" : "${var.skucapacity}",
          "tier" : "${var.skutier}"
        },
        "version" : "${var.pgversion}"
      }
    }
  }
}`

`{
  "resource" : {
    "azurerm_postgresql_database" : {
      "testdb" : {
        "charset" : "UTF8",
        "server_name" : "${azurerm_postgresql_server.pgserver.name}",
        "name" : "${var.pgDBs[2]}",
        "resource_group_name" : "${var.resource_group_name}",
        "collation" : "en_US"
      }
    }
  }
}`

### Debug Output

<!---
 Error: Error applying plan:
1 error(s) occurred:
* azurerm_postgresql_database.testdb (destroy): 1 error(s) occurred:
* azurerm_postgresql_database.testdb: Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: '5d3cfe7f-d576-4046-a48e-3f587064ed13'"
Terraform does not automatically rollback in the face of errors.

--->

### Expected Behavior

It should destroy the resources from Azure

### Actual Behavior

 Error: Error applying plan:
1 error(s) occurred:
* azurerm_postgresql_database.testdb (destroy): 1 error(s) occurred:
* azurerm_postgresql_database.testdb: Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: '5d3cfe7f-d576-4046-a48e-3f587064ed13'"
Terraform does not automatically rollback in the face of errors.


### Steps to Reproduce

1. `terraform apply`

2. `terraform destroy`

Hi @tombuildsstuff 

we can reproduce this issue for a while in `westeurope` region. According to some tests this seems only be the case if we create a azurerm_postgresql_server in combination with azurerm_sql_database (but not 100% sure if this is releated). Agree with you this seems to be related to Azure, so we will open a Support ticket there. 

```
TestPRV_Postgres 2019-10-10T11:46:02Z command.go:158: module.postgres.azurerm_postgresql_database.pg_db["unit-tests-tfm-azure-postgres-db-1"]: Creating...
TestPRV_Postgres 2019-10-10T11:46:12Z command.go:158: module.postgres.azurerm_postgresql_database.pg_db["unit-tests-tfm-azure-postgres-db-1"]: Still creating... [10s elapsed]
TestPRV_Postgres 2019-10-10T11:46:18Z command.go:158: 
TestPRV_Postgres 2019-10-10T11:46:18Z command.go:158: Error: Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: '36de0215-189f-482e-9be7-c03588812ea4'"
TestPRV_Postgres 2019-10-10T11:46:18Z command.go:158: 
TestPRV_Postgres 2019-10-10T11:46:18Z command.go:158:   on ../main.tf line 27, in resource "azurerm_postgresql_database" "pg_db":
TestPRV_Postgres 2019-10-10T11:46:18Z command.go:158:   27: resource "azurerm_postgresql_database" "pg_db" {
TestPRV_Postgres 2019-10-10T11:46:18Z command.go:158: 
TestPRV_Postgres 2019-10-10T11:46:18Z command.go:158: 
```

--- a/azurerm/helpers/validate/database.go
+++ b/azurerm/helpers/validate/database.go
@@ -15,7 +15,7 @@ func DatabaseCollation(i interface{}, k string) (warnings []string, errors []err
 	matched, _ := regexp.MatchString(`^[-A-Za-z0-9_. ]+$`, v)
 
 	if !matched {
-		errors = append(errors, fmt.Errorf("%s contains invalid characters, only underscores are supported, got %s", k, v))
+		errors = append(errors, fmt.Errorf("%s contains invalid characters, only alphanumeric, underscores, spaces or hyphens characters are supported, got %s", k, v))
 		return
 	}
 

--- a/azurerm/helpers/validate/database.go
+++ b/azurerm/helpers/validate/database.go
@@ -15,7 +15,7 @@ func DatabaseCollation(i interface{}, k string) (warnings []string, errors []err
 	matched, _ := regexp.MatchString(`^[-A-Za-z0-9_. ]+$`, v)
 
 	if !matched {
-		errors = append(errors, fmt.Errorf("%s contains invalid characters, only alphanumeric, underscores, spaces or hyphens characters are supported, got %s", k, v))
+		errors = append(errors, fmt.Errorf("%s contains invalid characters, only alphanumeric, underscore, space or hyphen characters are supported, got %s", k, v))
 		return
 	}
 

--- a/azurerm/helpers/validate/database.go
+++ b/azurerm/helpers/validate/database.go
@@ -12,7 +12,7 @@ func DatabaseCollation(i interface{}, k string) (warnings []string, errors []err
 		return
 	}
 
-	matched, _ := regexp.MatchString(`^[A-Za-z0-9_. ]+$`, v)
+	matched, _ := regexp.MatchString(`^[-A-Za-z0-9_. ]+$`, v)
 
 	if !matched {
 		errors = append(errors, fmt.Errorf("%s contains invalid characters, only underscores are supported, got %s", k, v))

--- a/azurerm/helpers/validate/database_test.go
+++ b/azurerm/helpers/validate/database_test.go
@@ -8,6 +8,10 @@ func TestDatabaseCollation(t *testing.T) {
 		Errors int
 	}{
 		{
+			Value:  "en@US",
+			Errors: 1,
+		},
+		{
 			Value:  "en-US",
 			Errors: 0,
 		},

--- a/azurerm/helpers/validate/database_test.go
+++ b/azurerm/helpers/validate/database_test.go
@@ -9,7 +9,7 @@ func TestDatabaseCollation(t *testing.T) {
 	}{
 		{
 			Value:  "en-US",
-			Errors: 1,
+			Errors: 0,
 		},
 		{
 			Value:  "en_US",

--- a/azurerm/resource_arm_postgresql_database.go
+++ b/azurerm/resource_arm_postgresql_database.go
@@ -152,11 +152,7 @@ func resourceArmPostgreSQLDatabaseRead(d *schema.ResourceData, meta interface{})
 
 	if props := resp.DatabaseProperties; props != nil {
 		d.Set("charset", props.Charset)
-
-		if collation := props.Collation; collation != nil {
-			v := strings.Replace(*collation, "-", "_", -1)
-			d.Set("collation", v)
-		}
+		d.Set("collation", props.Collation)
 	}
 
 	return nil

--- a/azurerm/resource_arm_postgresql_database.go
+++ b/azurerm/resource_arm_postgresql_database.go
@@ -3,7 +3,6 @@ package azurerm
 import (
 	"fmt"
 	"log"
-	"strings"
 	"time"
 
 	"github.com/Azure/azure-sdk-for-go/services/postgresql/mgmt/2017-12-01/postgresql"

